linker/Mono.Linker/Annotations.cs

```diff
@@ -349,11 +349,40 @@ public void Pop ()
 			dependency_stack.Pop ();
 		}
 
+		bool IsNestedInPrivate (TypeDefinition td)
```

Comment:
I am not sure what is the intention here, what is nested in private?
You should also move td.IsNestedPrivate check here to make this work with 2+ levels of nesting

Reply:
I have added explanation to the commit message and updated the patch.